pkg/rehearse/rehearse.go

```diff
@@ -630,14 +639,38 @@ func ensureImageStreamTags(ctx context.Context, client ctrlruntimeclient.Client,
 				istLog.WithError(err).Errorf("failed waiting for imagestreamtag to appear")
 				return fmt.Errorf("failed waiting for imagestreamtag %s to appear: %w", requiredImageStreamTag, err)
 			}
-
 			return nil
 		})
 	}
 
 	return g.Wait()
 }
 
+func ensureISTSInQCI(ctx context.Context, client ctrlruntimeclient.Client,
+	ists apihelper.ImageStreamTagMap, registryConfig string, log *logrus.Entry) error {
+
+	ocImageInfoOptions := quayiociimagesdistributor.OCImageInfoOptions{
+		RegistryConfig: registryConfig,
+		// TODO: multi-arch support
+		FilterByOS: "linux/amd64",
+	}
+	ocClientFactory := quayiociimagesdistributor.NewClientFactory()
+	quayIOImageHelper, err := ocClientFactory.NewClient()
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to create QuayIOImageHelper")
+	}
+
+	var errs []error
+	for _, ist := range ists {
+		requiredImageStreamTag := ist
+		if err := quayiociimagesdistributor.EnsureISTinQCI(ctx, client, requiredImageStreamTag, quayIOImageHelper,
```

Comment:
Could we do the mirror right away with `quayIOImageHelper`?
I think we do not even need to check if it exists on QCI already or is out of date.
Moreover, the nil case of store wont work at the moment, because we never consume the task in the store.